python-threatexchange/threatexchange/hasher/hasher.py

```diff
@@ -0,0 +1,36 @@
+import pathlib
+import typing as t
+from threatexchange.cli.cli_config import CLISettings
+
+from threatexchange.content_type.content_base import ContentType
+from threatexchange.signal_type.signal_base import FileHasher, SignalType
+
+
+class Hasher:
+    def __init__(
+        self,
+        settings: CLISettings,
```

Comment:
It looks like the only thing you pass in the settings type for is the list of signal types. What if you passed in the full list of signal types? 

Alternatively, you only use content type to narrow down to the applicable signal types. What if you only passed in those?